src/ert/config/observations.py

```diff
@@ -1,471 +1,243 @@
-        value = observation_dict.value
-        return (
-            value,
-            float(
-                EnkfObs._handle_error_mode(
-                    np.array(value),
-                    observation_dict,
-                )
-            ),
-        )
-
-    @classmethod
-    def _handle_summary_observation(
-        cls,
-        summary_dict: SummaryValues,
-        obs_key: str,
-        time_map: List[datetime],
-        has_refcase: bool,
-    ) -> Dict[str, ObsVector]:
-        summary_key = summary_dict.key
-        value, std_dev = cls._make_value_and_std_dev(summary_dict)
-
-        try:
-            if summary_dict.date is not None and not time_map:
-                # We special case when the user has provided date in SUMMARY_OBS
-                # and not REFCASE or time_map so that we dont change current behavior.
-                try:
-                    date = datetime.fromisoformat(summary_dict.date)
-                except ValueError as err:
-                    raise ValueError("Please use ISO date format YYYY-MM-DD.") from err
-                restart = None
-            else:
-                restart = cls._get_restart(summary_dict, obs_key, time_map, has_refcase)
-                date = time_map[restart]
-        except ValueError as err:
-            raise ObservationConfigError.with_context(
-                f"Problem with date in summary observation {obs_key}: " + str(err),
-                obs_key,
-            ) from err
-
-        if restart == 0:
-            raise ObservationConfigError.with_context(
-                "It is unfortunately not possible to use summary "
-                "observations from the start of the simulation. "
-                f"Problem with observation {obs_key}"
-                f"{' at ' + str(cls._get_time(summary_dict, time_map[0])) if summary_dict.restart is None else ''}",
-                obs_key,
-            )
-        return {
-            obs_key: ObsVector(
-                EnkfObservationImplementationType.SUMMARY_OBS,
-                summary_key,
-                "summary",
-                {date: SummaryObservation(summary_key, obs_key, value, std_dev)},
-            )
-        }
-
-    @classmethod
-    def _create_gen_obs(
-        cls,
-        scalar_value: Optional[Tuple[float, float]] = None,
-        obs_file: Optional[str] = None,
-        data_index: Optional[str] = None,
-    ) -> GenObservation:
-        if scalar_value is None and obs_file is None:
-            raise ValueError(
-                "Exactly one the scalar_value and obs_file arguments must be present"
-            )
+    def __repr__(self) -> str:
+        return f"EnkfObs({self.obs_vectors}, {self.obs_time})"
 
-        if scalar_value is not None and obs_file is not None:
-            raise ValueError(
-                "Exactly one the scalar_value and obs_file arguments must be present"
-            )
+    def get_dataset(self, key: str) -> Optional[xr.Dataset]:
```

Comment:
Can you create an issue for rewriting this? Ok to leave it for this PR, but better to have more coverage of production code, then having functionality just for tests.

Reply:
This is the only test running the code:
```def test_migrate_observations(setup_case, set_ert_config):
    ert_config = setup_case("block_storage/version-3/poly_example", "poly.ert")
    with open_storage(ert_config.ens_path, "w") as storage:
        assert len(list(storage.experiments)) == 1
        experiment = list(storage.experiments)[0]
        assert experiment.observations == {
            k: ert_config.observations.get_dataset(k).squeeze("name", drop=True)
            for k in experiment.observations
        }
```
Maybe just remove the test? Issue would be basically "remove the test" as the behavior it asserts is a bit "deprecated"

Reply:
As long as we have a test that covers the migration, then deleting that one is ok. If there is not, it should be rewritten and get a better assert.